File: baekjoon/python/11725/11725.py
from collections import deque
import sys
sys.stdin = open('11725_input.txt', 'r')

n = int(input())

graph = [[] for _ in range(n+1)]

print(graph)

for i in range(n-1):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

print(graph)

visited = [False]*(n+1)
parent = [0]*(n+1)


def bfs(root):
    q = deque()
    q.append(root)
    while q:
        p = q.popleft()
        if not visited[p]:
            visited[p] = True
            for nv in graph[p]:
                parent[nv] = p
                q.append(nv)


bfs(1)
print(parent)

File: baekjoon/cpp/11725/11725.cpp
#include <bits/stdc++.h>
using namespace std;
vector<int> graph[100001];
int main()
{
    std::ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    freopen("11725_input.txt", "r", stdin);

    int n;
    cin >> n;
    for (int i = 1; i < n; i++)
    {
        int a, b;
        cin >> a >> b;
        graph[a].push_back(b);
        graph[b].push_back(a);
    }
    for (int i = 1; i <= n; i++)
    {
        for (auto &elem : graph[i])
        {
            cout << elem << " ";
        }
        cout << endl;
    }

    return 0;
}
